### ISSUE TITLE
How to convert my own video training dataset to your type TFrecords?

### ISSUE THREAD

**Issue by OliviaMG** [tensorflow/models#579]
## Please let us know which model this issue is about (specify the top-level directory)
Hi, Thanks very much first for sharing the code
I want to use your code (video prediction) in my work. Instead of using push dataset, I have my own dataset, videos. I used below method to convert the video to TFrecords file, but after i run the training.py, the only thing show to me is gibberish.

Could you mind share me the code that you used to convert the video to TFrecords?

Thanks very much in advance.

########my code #############
import os, sys
import tensorflow as tf
import numpy as np
from PIL import Image

def _int64_feature(value):
return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))

def _bytes_feature(value):
return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))

def read():

# parent_foler contians all sub-folder, each sub-folder has a sequence of images
parent_foler = sys.argv[1]

# sub-folder
for folder in os.listdir(parent_foler):
    if os.path.isfile(folder): continue

    label = folder.split('_')[0].strip()

    print label

    if label == 'grasp':
        label = 1
    if label == 'up':
        label = 2
    if label == 'down':
        label = 3
    if label == 'left':
        label = 4
    if label == 'right':
        label = 5
    if label == 'forward':
        label = 6
    if label == 'backward':
        label = 7
    if type(label) == 'str':
        print 'label error'
        exit()

    print 'Reading: ' + folder

    # tf-record format image
    images = read_images_from(parent_foler + '/' +folder)
    num_examples = len(images)
    # rows = images.shape[1]
    # cols = images.shape[2]
    # depth = images.shape[3]

    print 'Number of images: ' + str(num_examples)

    outputFile = os.path.join(parent_foler, folder + '.tfrecord')

    writer = tf.python_io.TFRecordWriter(outputFile)

    fs = {}
    for index in range(num_examples):
        image_raw = images[index].tostring()
        # print image_raw
        image_name = 'move/' + str(index) + '/image/encoded'
        fs[image_name] = _bytes_feature(image_raw)
        print folder + ':' + image_name
    print 'Size of Features:' + str(len(fs))
    example = tf.train.Example(features=tf.train.Features(feature=fs))
    writer.write(example.SerializeToString())
    writer.close()
def convert_to_image_per_folder(folder_name):

files_to_read = [folder_name + '/' + folder_name.split('/')[-1] + '_' + str(i+1) + '.png' for i in range(10)]

# print files_to_read
filename_queue = tf.train.string_input_producer(files_to_read)

reader = tf.WholeFileReader()
key, value = reader.read(filename_queue)
my_img = tf.image.decode_png(value)  # use png or jpg decoder based on your files.

init_op = tf.initialize_all_variables()
sess = tf.InteractiveSession()
with sess.as_default():
    sess.run(init_op)

coord = tf.train.Coordinator()
threads = tf.train.start_queue_runners(coord=coord)

image = []
for i in range(len(files_to_read)):
    image.append(my_img.eval(session=sess))

coord.request_stop()
coord.join(threads)
return image
def read_images_from(folder_name):

images = []
files_to_read = [folder_name + '/' + folder_name.split('/')[-1] + '_' + str(i + 1) + '.png' for i in range(10)]

for filename in files_to_read:
im = Image.open(filename)
im = np.asarray(im, np.uint8)
images.append(im)

# images_only = [np.asarray(image[1], np.uint8) for image in images] # Use unint8 or you will be !!!
images_only = np.array(images)

print(images_only.shape)
return images_only

if name == "main":
read()

###############error############

��������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������~���~���|���|���~�������������������������������������������������������������������������������������������������������������������������������������������~���������������}�������������������������������������������������������������������������������������������������������������������������������������������������������~���}���}���~�������������������������������~���~���~���~���~���}��~|�~}{�~}{�~}{�~}{�~}{��~|���}���}���}���~���~���|���|���|���|���}���~���}���{�~}y�}|x�|{w�|{w�}|x�|{w�{zv�zyu�xws�wvr�vuq�utp�tso�tso�tso�utp�tso�utp�utp�utp�tso�tso�utp�utp�tso�utp�utp�vuq�wvr�xws�yxt�zyu�{zv�|{w�}|x�}|x�}|x�}|x�}|x�}|x�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}y�~}x��~y��~w��~w��~w��~w��~w��~w��~w��~w��~w��~W tensorflow/core/framework/op_kernel.cc:968] Invalid argument: Could not parse example input, value: '
���<
����
�move/7/image/encoded�����
����
@cbfinn

**Comment by drpngx** [comment]
Hi, this sounds like it would be a good fit for a stackoverflow question. Please try there (add the tag "tensorflow" so we can follow it).

**Comment by cbfinn** [comment]
See example code, posted by Frederik, here: https://github.com/OliviaMG/xiaomeng/issues/1

**Comment by 386083520** [comment]
import os, sys
import tensorflow as tf
import numpy as np
from PIL import Image

def _int64_feature(value):
    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))

def _bytes_feature(value):
    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))

def read():
    # parent folder contains all sequence, each sequence (10 png) is in a sub-folder
    parent_foler = "push/push_testnovel2"
    normalFileName="I:\python\pycharm\\video_prediction-master\\video_prediction-master\push\origiondata\men"
    for folder in os.listdir(normalFileName):
        images = read_images_from(normalFileName + '\\' + folder)
        num_examples = len(images)
        print ('Number of images: ' + str(num_examples))
        outputFile = os.path.join(parent_foler, folder + '.tfrecords')
        writer = tf.python_io.TFRecordWriter(outputFile)
        fs = {}
        for index in range(num_examples):
            image_raw = images[index].tostring()
            image_name = 'move/' + str(index) + '/image/encoded'
            fs[image_name] = _bytes_feature(image_raw)
            print (folder + ':' + image_name)
        print ('Size of Features:' + str(len(fs)))
        example = tf.train.Example(features=tf.train.Features(feature=fs))
        writer.write(example.SerializeToString())
        writer.close()

def read_images_from(folder_name):
    images = []
    files_to_read = [folder_name + "\\" + str(i) + '.jpeg' for i in range(10)]
    for filename in files_to_read:
        im = Image.open(filename)
        im = np.asarray(im, np.uint8)
        images.append(im)
    images = np.array(images)
    print('shape of images: ' + str(images.shape))
    return images

if __name__ == "__main__":
    read()

Why  the data set generated by this code can not be used.

**Comment by 386083520** [comment]
Is someone can provide a correct code to generate the tfrecords dataset?I have met a lot of trouble
